fractal_server/app/routes/admin/v2.py

```diff
@@ -307,3 +314,126 @@ async def flag_task_v1_as_v2_compatible(
     await db.close()
 
     return Response(status_code=status.HTTP_200_OK)
+
+
+class TaskV2Minimal(BaseModel):
+
+    id: int
+    name: str
+
+    type: str
+    command_non_parallel: Optional[str] = None
+    command_parallel: Optional[str] = None
+
+    source: str
+
+    meta_non_parallel: dict[str, Any] = Field(default_factory=dict)
+    meta_parallel: dict[str, Any] = Field(default_factory=dict)
+
+    owner: Optional[str]
+    version: Optional[str]
+
+    docs_info: Optional[str] = None
+    docs_link: Optional[HttpUrl] = None
+
+    input_types: dict[str, bool] = Field(default_factory=dict)
+    output_types: dict[str, bool] = Field(default_factory=dict)
+
+
+class ProjectUser(BaseModel):
+    id: int
+    email: EmailStr
+
+
+class TaskContext(BaseModel):
+
+    task_v2_minimal: TaskV2Minimal
+
+    workflow_id: Optional[int]
+    workflow_name: Optional[str]
+    project_id: Optional[int]
+    project_name: Optional[str]
+
+    project_users: Optional[list[ProjectUser]]
+
+
+@router_admin_v2.get("/task/", response_model=list[TaskContext])
+async def query_tasks(
+    id: Optional[int] = None,
+    source: Optional[str] = None,
+    version: Optional[str] = None,
+    name: Optional[str] = None,
+    owner: Optional[str] = None,
+    only_no_owner: bool = False,
```

Comment:
Let's turn this into a `kind: Optional[Literal["common", "users"]] = None`.
* `common` means that `owner` must be `None`
* `users` means that `owner` must be not `None`